extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -543,55 +517,53 @@ public long getAsLong() {
                 writeInstructions.getParquetColumnNameFromColumnNameOrDefault(name))) {
             boolean usedDictionary = false;
             if (supportsDictionary(columnSource.getType())) {
-                usedDictionary = tryEncodeDictionary(columnDefinition,
-                        writeInstructions,
+                usedDictionary = tryEncodeDictionary(writeInstructions,
+                        tableRowSet,
                         rowSet,
+                        columnDefinition,
+                        columnWriter,
                         columnSource,
                         lengthSource,
-                        lengthRowSet,
-                        targetRowsPerPage,
                         rowStepGetter,
                         valuesStepGetter,
-                        pageCount,
-                        columnWriter);
+                        targetRowsPerPage,
```

Comment:
I'm decently sure you need to compute the max rows per page, and that is what you should be using to size the data `GetContext` in the `encodeX` methods. I thought that was the intent behind your "escape hatch", which you removed entirely instead of just using it as a place to record max data page size.